v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/PipelineController.java

```diff
@@ -158,6 +142,43 @@ static PipelineResult executeSingleInstanceMigration(
     return pipeline.run();
   }
 
+  @NotNull
+  /**
+   * This list will contain the final list of tables that actually get migrated, which will be the
+   * intersection of Spanner and source tables.
+   */
+  static List<String> listSpannerTablesToMigrate(
+      Ddl ddl, ISchemaMapper schemaMapper, Set<String> tablesToMigrateSet) {
+    // This list is all Spanner tables topologically ordered.
+    List<String> orderedSpTables = ddl.getTablesOrderedByReference();
+
+    // This list will contain the final list of tables that actually get migrated, which will be the
+    // intersection of Spanner and source tables.
+    List<String> finalTablesToMigrate = new ArrayList<>();
+    for (String spTable : orderedSpTables) {
+      try {
+        String srcTable = schemaMapper.getSourceTableName("", spTable);
+        if (!tablesToMigrateSet.contains(srcTable)) {
+          continue;
+        }
+        finalTablesToMigrate.add(spTable);
+      } catch (NoSuchElementException e) {
+        LOG.info("ignoring table not identified by schema mapper: {}", spTable);
```

Comment:
Confirming, the only case that was not handled was when there was a session file involved and Spanner had extra table that were not in the session file and its this try catch protecting us from that now?

Reply:
yes. This is to handle scenarios where the mapper does not have reference to the table.